File: Easy/ShuffleTheArray.js
/*
Given the array nums consisting of 2n elements in the form [x1,x2,...,xn,y1,y2,...,yn].

Return the array in the form [x1,y1,x2,y2,...,xn,yn].
*/


/**
 * @param {number[]} nums
 * @param {number} n
 * @return {number[]}
 */
var shuffle = function(nums, n) {
    let arr = [];
    for (let i = 0; i < n; i++) {
        arr.push(nums[i], nums[i + n]);
    }
    return arr;
};

File: Easy/shuffle_the_array.cpp
class Solution {
public:
    vector<int> shuffle(vector<int>& nums, int n) {
        for(int i = n; i<nums.size(); i++){
            nums[i] = (nums[i] * 1024) + nums[i-n];
        }
        
        for(int i = n, index = 0; i < nums.size(); i++, index+=2){
            nums[index] = nums[i] % 1024;
            nums[index+1] = nums[i] / 1024;
        }
        return nums;
    }
};